spec/ddtrace/metrics_spec.rb

```diff
@@ -246,10 +270,17 @@
     subject(:default_statsd_client) { metrics.default_statsd_client }
 
     let(:statsd_client) { instance_double(Datadog::Statsd) }
+    let(:options) do
+      if Gem::Version.new(Datadog::Statsd::VERSION) >= Gem::Version.new('5.2.0')
+        { single_thread: true }
+      else
+        {}
+      end
+    end
 
     before do
       expect(Datadog::Statsd).to receive(:new)
-        .with(metrics.default_hostname, metrics.default_port)
+        .with(metrics.default_hostname, metrics.default_port, **options)
         .and_return(statsd_client)
     end
```

Comment:
Minor: May be worth leaving here a note that this integration-y test is ok because we run appraisals with both latest dogstatsd and the 4.x series.